app/main/util/search_utils.py

```diff
@@ -0,0 +1,174 @@
+from app.main.util.pagination import calculate_total_pages, get_pagination
+
+
+def get_param(param):
+    return request.form.get(param, "") or request.args.get(param, "")
+
+
+def get_query_and_search_area():
+    """Fetch query and search_area from form or request args"""
+    query = get_param("query")
+    search_area = get_param("search_area")
+    return query.strip(), search_area
+
+
+def setup_opensearch():
+    """Setup and return an OpenSearch client"""
+    return OpenSearch(
+        hosts=current_app.config.get("OPEN_SEARCH_HOST"),
+        http_auth=current_app.config.get("OPEN_SEARCH_HTTP_AUTH"),
+        use_ssl=True,
+        verify_certs=True,
+        ca_certs=current_app.config.get("OPEN_SEARCH_CA_CERTS"),
+        connection_class=RequestsHttpConnection,
+    )
+
+
+def execute_search(open_search, dsl_query, page, per_page):
+    """Execute the search query using OpenSearch"""
+    size = per_page
+    from_ = size * (page - 1)
+    try:
+        return open_search.search(
+            dsl_query,
+            from_=from_,
+            size=size,
+            timeout=current_app.config["OPEN_SEARCH_TIMEOUT"],
+        )
+    except opensearchpy.exceptions.ConnectionTimeout:
+        abort(504)
+
+
+def get_pagination_info(results, page, per_page):
+    """Calculate pagination information"""
+    total_records = (
+        results["hits"]["total"]["value"] if "hits" in results else 0
+    )
+    page_count = calculate_total_pages(total_records, per_page)
+    pagination = get_pagination(page, page_count)
+    return total_records, pagination
+
+
+def get_filtered_list(to_filter, filter):
+    """Filters a list based on the contents of another list"""
+    return [field for field in to_filter if field not in filter or []]
+
+
+def get_all_fields_excluding(open_search, index_name, exclude_fields=None):
+    """Retrieve all fields from the index and exclude certain fields"""
+    mappings = open_search.indices.get_mapping(index=index_name)
+    all_fields = list(mappings[index_name]["mappings"]["properties"].keys())
+    filtered_fields = get_filtered_list(all_fields, exclude_fields)
+
+    return filtered_fields
+
+
+def get_query_aggregations():
+    """Returns the aggregations segment of the DSL query"""
+    return {
+        "aggs": {
+            "aggregate_by_transferring_body": {
+                "terms": {"field": "transferring_body_id.keyword"},
+                "aggs": {
+                    "top_transferring_body_hits": {
+                        "top_hits": {
+                            "size": 1,
+                            "_source": ["transferring_body"],
+                        }
+                    }
+                },
+            }
+        },
+    }
+
+
+def get_query_multi_match(query, search_area, open_search, _id):
+    """Returns the multi-match segment of the DSL query"""
+
+    fields_record = ["file_name", "file_path", "content"]
```

Comment:
thinking we can move this block from 93 to 102, defining the `fields` variable outside this function and then pass it in as an argument, that way we don't need to pass open_search instance to the build_dsl_query at all, then it will just be an easily tested function without any mocking.

Then we just need to do extract this out, maybe to its own function, something like `get_open_search_fields_to_search_on` and call it before we call this function in the routes.

Then the test for the new function is the only test that needs to mock opensearch (apart from the route test)